src/kademlia/routing_table.cpp

```diff
@@ -682,13 +701,37 @@ routing_table::add_node_status_t routing_table::add_node_impl(node_entry e)
 		else
 		{
 			TORRENT_ASSERT(existing->id != e.id);
-			// this is the same IP and port, but with
-			// a new node ID. remove the old entry and
-			// replace it with this new ID
+			// This is the same IP and port, but with a new node ID.
+			// This may indicate a malicious node so remove the entry.
+#ifndef TORRENT_DISABLE_LOGGING
+			char hex_id_new[41];
+			char hex_id_old[41];
+			aux::to_hex(e.id.data(), 20, hex_id_new);
+			aux::to_hex(existing->id.data(), 20, hex_id_old);
+			m_log->log(dht_logger::routing_table, "evicting node (changed ID): old: %s new: %s %s"
```

Comment:
If m_log being NULL was the problem I would have expected it to fail consistently on all platforms, but I can't reproduce it. In any case I've pushed a fix to always check m_log.